docs/guests.md

```diff
@@ -53,6 +53,27 @@ rc-service xe-guest-utilities start
 apt install xe-guest-utilities
 ```
 
+#### openSUSE Leap 15.2 with transactional-updates
+```
+sudo transactional-updates shell
+
+```
+inside the chroot enviroment
+
+```
+mount /dev/cdrom /mnt
+bash /mnt/Linux/install.sh -d sles -m 15
+umount /dev/cdrom
+```
+then kill the xe-daemon (important), exit the chroot and reboot.
```

Comment:
Please explain why it's important to kill the xe-daemon. And how can it be running if insserv hasn't been installed already? The service wouldn't start. Maybe you should talk  about insserv higher in your guide, as a prerequisite.

Reply:
Preface: The idea behind transactional updates is that you can automatically apply patches and installs without the risk of braking the system. To do that the openSUSE leverages btrfs and mounts the root file system as read only and runs the script transactional-updates, which opens up chroot on a snapshot of the root file system with writes enabled. If the script terminates without failures the update was successful and and you can mount the snapshot in the next reboot.

I found that xe-daemon binds to /proc/xen (don't quote me on this  I'm not sure on where exactly in proc) and a few others which it doesn't let go when exiting the chroot and closing the snapshot. So transactional updates fail. I tried a lot of different approaches to install xen tools in other ways but without much luck and the were all way more hacky than just killing the daemon manually.


> And how can it be running if insserv hasn't been installed already?
I'm not sure how the daemon is started in the first place from the rpm so your guesses are as good as mine.

> The service wouldn't start.
It seams that it is not the systemd service that starts cause we are in a chroot, where Systemd is not reachable. At least to my knowledge.

So my speculation is that the rpm try's to start the daemon in different ways. Maybe it detects the absence of systemd and uses other ways to start up. I didn't look that far into it.

> Maybe you should talk about insserv higher in your guide, as a prerequisite.
That is a good suggestion was not happy to put it at the end but was the only place I thought of.